university/src/main/java/nl/quintor/studybits/university/config/UniversitiesBeanConfig.java

```diff
@@ -19,81 +20,84 @@
 @Configuration
 public class UniversitiesBeanConfig {
 
-    @Autowired
-    private IndyPool indyPool;
-
-    @Autowired
-    @Qualifier("stewardTrustAnchor")
-    private TrustAnchor steward;
-
-    @Bean("RugUniversity")
-    public Issuer rugUniversity() {
-        return ensureIssuer("Rug");
-    }
-
-    @Bean("GentUniversity")
-    public Issuer gentUniversity() {
-        return ensureIssuer("Gent");
-    }
-
-    private Issuer ensureIssuer(String universityName) {
-        return createIssuer(universityName)
-                .orElseGet(() -> getIssuer(universityName));
-    }
-
-    private Optional<Issuer> createIssuer(String name) {
-        IndyWallet wallet = createIndyWallet(name);
-        return wallet != null ? Optional.of(onboardIssuer(wallet)) : Optional.empty();
-    }
-
-    private Issuer getIssuer(String name) {
-        IndyWallet wallet = getIndyWallet(name);
-        return new Issuer(name, indyPool, wallet);
-    }
-
-    @SneakyThrows
-    private Issuer onboardIssuer(IndyWallet wallet) {
-        Issuer issuer = new Issuer(wallet.getName(), indyPool, wallet);
-        // Connect issuer with university
-        ConnectionRequest connectionRequest = steward
-                .createConnectionRequest(issuer.getName(), "TRUST_ANCHOR")
-                .get();
-        AnoncryptedMessage newcomerConnectionResponse = issuer
-                .acceptConnectionRequest(connectionRequest)
-                .thenCompose(AsyncUtil.wrapException(issuer::anoncrypt))
-                .get();
-        steward.anonDecrypt(newcomerConnectionResponse, ConnectionResponse.class)
-                .thenCompose(AsyncUtil.wrapException(steward::acceptConnectionResponse))
-                .get();
-        AuthcryptedMessage verinym = issuer
-                .authcrypt(issuer.createVerinymRequest(connectionRequest.getDid()))
-                .get();
-        steward.authDecrypt(verinym, Verinym.class)
-                .thenCompose(AsyncUtil.wrapException(steward::acceptVerinymRequest))
-                .get();
-        issuer.init();
-        return issuer;
-    }
-
-    @SneakyThrows
-    private IndyWallet getIndyWallet(String name) {
-        return new IndyWallet(name);
-    }
-
-    @SneakyThrows
-    private IndyWallet createIndyWallet(String name) {
-        try {
-            return IndyWallet.create(indyPool, name, createSeed(name));
-        } catch (Exception ex) {
-            if (ex.getCause() instanceof WalletExistsException) {
-                return null;
-            }
-            throw ex;
-        }
-
-    }
-
-    private String createSeed(String s) {
-        return StringUtils.leftPad(s, 32, '0');
-    }
+//    private final IndyPool indyPool;
```

Comment:
Is this meant to be commented?

Reply:
This complete file should be removed. The UniversityService will now create the IndyWallet and Issuer when a new University is added. This means there is no limitation of number of universities  defined as beans.